programs/client/Client.cpp

```diff
@@ -131,69 +132,64 @@ void Client::showWarnings()
     }
 }
 
-void Client::parseConnectionsCredentials()
+void Client::parseConnectionsCredentials(Poco::Util::AbstractConfiguration & config, const std::string & name)
```

Comment:
Should here be a check
```
if (!config.has("connections_credentials"))
    return;
```

Reply:
Here we have different `config`, that contains only the loaded xml/yaml, so it cannot be done this way.

Reply:
But below you get it from this config (line 155):
```
config.keys("connections_credentials", keys);
```
so I was referring to an early return or exception

Reply:
Looks like an early return can be 
```
if (!config.has("connections_credentials")) 
{
    if (name.empty())
        return;
    throw ...
}
```

Reply:
Oh, you are right, I misread your first comment, as `if (!config.has("connection"))` (that is from the cli args). But still, `Client::parseConnectionsCredentials` should be called exactly on the config from file, not from the CLI options, CLI options should always have preference over config values, hence the exception is throwed here:

```c
if (config_path.has_value())
{
    ConfigProcessor config_processor(*config_path);
    auto loaded_config = config_processor.loadConfig();
    parseConnectionsCredentials(*loaded_config.configuration, config().getString("connection", ""));
    config().add(loaded_config.configuration);
}
else if (config().has("connection"))
    throw Exception(ErrorCodes::BAD_ARGUMENTS, "--connection was specified, but config does not exists");
```

Reply:
yeah, I agree, I did not suggest to change the logic, but just adding an early return inside `parseConnectionsCredentials`,  where `config` in my example is `*loaded_config.configuration` so no logic is changed (may be you thought in my example `config` is `config()`, but it is `loaded_config.configuration`?). This change is not necessary, just a suggestion as it seems to make the purpose of this `parseConnectionsCredentials` method understood without reading it all through.

Reply:
Actually even this cannot be done, since right now there is a fallback to use hostname as the default connection name, so let's leave this as-is.